### PR TITLE
[SPARK-15428][SQL] Disable multiple streaming aggregations

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?

Incrementalizing plans of with multiple streaming aggregation is tricky and we dont have the necessary support for "delta" to implement correctly. So disabling the support for multiple streaming aggregations.
## How was this patch tested?

Additional unit tests

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala [hunk 1]
```diff
@@ -55,10 +55,19 @@ object UnsupportedOperationChecker {
...
+            throwError(
+              "Aggregations are not supported on streaming DataFrames/Datasets in " +
+                "Append output mode. Consider changing output mode to Update.")
+          }
+          val moreStreamingAggregates = child.find {
+            case Aggregate(_, _, grandchild) if grandchild.isStreaming => true
+            case _ => false
+          }
+          if (moreStreamingAggregates.nonEmpty) {
+            throwError("Multiple streaming aggregations are not supported DataFrames/Datasets")
```
Comment (jaceklaskowski) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala]: "are not supported DataFrames/Datasets"? It appears as "in" or "for" might be missing?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala [hunk 2]
```diff
@@ -55,10 +55,19 @@ object UnsupportedOperationChecker {
...
 
-        case Aggregate(_, _, child) if child.isStreaming && outputMode == Append =>
-          throwError(
-            "Aggregations are not supported on streaming DataFrames/Datasets in " +
-              "Append output mode. Consider changing output mode to Update.")
+        case Aggregate(_, _, child) if child.isStreaming =>
+          if (outputMode == Append) {
+            throwError(
+              "Aggregations are not supported on streaming DataFrames/Datasets in " +
+                "Append output mode. Consider changing output mode to Update.")
```
Comment (jaceklaskowski) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala]: I'd suggest changing "Append" to using the Append value and Update too, so when you refactor the code it'd be easier to catch.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala [hunk 3]
```diff
@@ -55,10 +55,19 @@ object UnsupportedOperationChecker {
...
+          }
+          val moreStreamingAggregates = child.find {
+            case Aggregate(_, _, grandchild) if grandchild.isStreaming => true
+            case _ => false
+          }
+          if (moreStreamingAggregates.nonEmpty) {
```
Comment (jaceklaskowski) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala]: What is the type of `moreStreamingAggregates`? Option? What do you think about `foreach`?

Comment (tdas) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala]: TreeNode.find returns an Option.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala (hunk 1 above)
Comment (tdas) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala]: will fix. thanks!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala (hunk 2 above)
Comment (tdas) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala]: I didnt get you. IntelliJ seems to catching all the uses of Append object properly.

**Comment by tdas** [comment]
cc @marmbrus

**Comment by marmbrus** [comment]
LGTM pending grammar fixes.

**Comment by tdas** [comment]
I am merging this to master and 2.0. Thank you for reviewing!

**Pull request closed without merging**